snowshu/core/docker.py

```diff
@@ -84,24 +98,61 @@ def get_stopped_container(  # noqa pylint: disable=too-many-arguments
         logger.info('Creating an external volume...')
         replica_volume = self._create_snowshu_volume(DOCKER_REPLICA_VOLUME)
 
-        logger.info(f"Creating stopped container {name}...")
-        container = self.client.containers.create(image,
-                                                  start_command,
-                                                  network=network.name,
-                                                  name=name,
-                                                  hostname=name,
-                                                  ports=port_dict,
-                                                  environment=envars,
-                                                  labels=labels,
-                                                  detach=True,
-                                                  volumes={replica_volume.name: {
-                                                      'bind': f'{DOCKER_REPLICA_MOUNT_FOLDER}',
-                                                      'mode': 'rw'
-                                                  }},
-                                                  working_dir=DOCKER_WORKING_DIR
-                                                  )
-        logger.info(f"Created stopped container {container.name}.")
-        return container
+        if not TARGET_ARCHITECTURE:
+            arch_list = [LOCAL_ARCHITECTURE]
+        else:
+            arch_list = TARGET_ARCHITECTURE
+
+        logger.info(f'Finding base image {image_name}...')
+        container_list = []
+        for arch in arch_list:
```

Comment:
(?) for the multi-arch case, the image-pull command should be executed in parallel 
can be  improved later

Reply:
Images are pulled as postgres:12 and that exact image instance is retagged to represent its arch, furthermore you cannot start those in parallel due to same ports.